macros.go

```diff
@@ -20,6 +22,87 @@ type MacroFunc func(*Query, []string) (string, error)
 // The "string" key is the name of the macro function. This name has to be regex friendly.
 type Macros map[string]MacroFunc
 
+// Default time filter for SQL based on the query time range.
+// It requires one argument, the time column to filter.
+// Example:
+//   $__timeFilter(time) => "time BETWEEN '2006-01-02T15:04:05Z07:00' AND '2006-01-02T15:04:05Z07:00'"
+func macroTimeFilter(query *Query, args []string) (string, error) {
+	if len(args) != 1 {
+		return "", errors.WithMessagef(ErrorBadArgumentCount, "expected 1 argument, received %d", len(args))
+	}
+
+	var (
+		column = args[0]
+		from   = query.TimeRange.From.UTC().Format(time.RFC3339)
+		to     = query.TimeRange.To.UTC().Format(time.RFC3339)
+	)
+
+	return fmt.Sprintf("%s BETWEEN '%s' AND '%s'", column, from, to), nil
```

Comment:
I don't think all the databases support `BETWEEN`.  Some require syntax like `timefield >= starttime AND timefield < endtime`. Wondering what is the most common approach.

Reply:
it seems to be [part of the standard](https://www.w3schools.com/sql/sql_between.asp) but that should also work.